### ISSUE TITLE
Auto updates never apply on macOS

### ISSUE THREAD

**Issue by sbrocket** [microsoft/vscode#100990]
Issue Type: <b>Bug</b>

I've had repeated difficulty getting VSCode updates to apply through the in-editor update system. For the last few updates, I've had to resort to manually downloading and updating the new version.

Steps:
1. When a new update is available (as it is currently, since I'm on 1.46.0), open Code and use "Check for updates..." from the menubar.
2. Update gets downloaded and the "Restart Visual Studio Code to apply the latest update." notification pops up in the lower right corner.
3. Click "Update Now".
4. VSCode exits but does not restart.
5. Manually restart VSCode, and note that it is still running the same version as before (1.46.0 currently).

These steps are 100% repeatable for me. There aren't any errors presented at any step in this process, the update just never happens. I don't see any errors like "Cannot update while running on a read-only volume. The application is on a read-only volume." that others have reporter on similar auto-updater issues. I've also tried the "xattr -dr com.apple.quarantine ~/Library/Caches/com.microsoft.VSCode.ShipIt* /Applications/Visual\ Studio\ Code.app/" workaround mentioned in other similar issues with no effect. This is on macOS 10.15.5.

VS Code version: Code 1.46.0 (a5d1cc28bb5da32ec67e86cc50f84c67cc690321, 2020-06-10T08:59:06.977Z)
OS version: Darwin x64 19.5.0
Remote OS version: Linux x64 5.2.17-1rodete3-amd64

<details>
<summary>System Info</summary>

|Item|Value|
|---|---|
|CPUs|Intel(R) Core(TM) i7-7700HQ CPU @ 2.80GHz (8 x 2800)|
|GPU Status|2d_canvas: enabled<br>flash_3d: enabled<br>flash_stage3d: enabled<br>flash_stage3d_baseline: enabled<br>gpu_compositing: enabled<br>metal: disabled_off<br>multiple_raster_threads: enabled_on<br>oop_rasterization: disabled_off<br>protected_video_decode: unavailable_off<br>rasterization: enabled<br>skia_renderer: disabled_off_ok<br>video_decode: enabled<br>viz_display_compositor: enabled_on<br>viz_hit_test_surface_layer: disabled_off_ok<br>webgl: enabled<br>webgl2: enabled|
|Load (avg)|4, 3, 3|
|Memory (System)|16.00GB (0.50GB free)|
|Process Argv||
|Screen Reader|no|
|VM|0%|

|Item|Value|
|---|---|
|Remote|SSH: desk|
|OS|Linux x64 5.2.17-1rodete3-amd64|
|CPUs|Intel(R) Xeon(R) CPU E5-2690 v4 @ 2.60GHz (56 x 1197)|
|Memory (System)|125.89GB (12.18GB free)|
|VM|0%|
</details><details><summary>Extensions (20)</summary>

Extension|Author (truncated)|Version
---|---|---
theme-monokai-pro-vscode|mon|1.1.17
vscode-json5|mrm|1.0.0
remote-containers|ms-|0.122.1
remote-ssh|ms-|0.51.0
remote-ssh-edit|ms-|0.51.0
remote-wsl|ms-|0.44.4
vscode-remote-extensionpack|ms-|0.20.0
vscode-gn|npc|1.1.0
vim|vsc|1.14.5
material-theme|zhu|3.8.2
githistory|don|0.6.5
nasc-touchbar|fel|1.12.0
language-fidl|fuc|0.3.1
Go|gol|0.14.4
vscode-clangd|llv|0.1.6
rust-analyzer|mat|0.2.216
git-graph|mhu|1.24.0
rewrap|stk|1.12.0
gitblame|wad|4.0.1
clang-format|xav|1.9.0

(2 theme extensions excluded)

</details>

**Comment by joaomoreno** [comment]
@sbrocket Thanks for the detailed issue description, you've done your homework! Can you try running `code --verbose` in a terminal and seeing if there are errors spewing when you click `Update Now`.

@deepak1556 Any more ideas?

**Comment by deepak1556** [comment]
Getting verbose logs is really helpful, along with that can you also add `--log-net-log=netlog.json` which will capture the network state, attach the log file once the application has quit.

**Comment by sbrocket** [comment]
Logs with both of those options attached.

[code-update-verbose.log](https://github.com/microsoft/vscode/files/4832544/code-update-verbose.log)

**Comment by deepak1556** [comment]
Thanks for the log, seems like the download was fine.

```
2020-06-25 09:12:11.808 Electron[50053:5893268] Download completed to: file:///Users/bryanhenry/Library/Caches/com.microsoft.VSCode.ShipIt/update.Ja0xMA8/VSCode-darwin-stable.zip
[90m[main 2020-06-25T16:12:13.816Z][0m update#setState ready
```

Can you attach the `ShipIt_stderr.log`, `ShipIt_stdout.log` under `Library/Caches/com.microsoft.VSCode.ShipIt/`

**Comment by sbrocket** [comment]
ShipIt_stdout.log is empty so it won't attach. ShipIt_stderr.log only seems to contain old logs from last year.

[ShipIt_stderr.log](https://github.com/microsoft/vscode/files/4832917/ShipIt_stderr.log)

**Comment by sbrocket** [comment]
On a whim I moved the old ShipIt_stdout.log and ShipIt_stderr.log out of that directory, because I noticed they were owned by root rather than my user and file permissions were set to 0644 (user writable only). After a few seconds VSCode opened with the update applied with no other action from me; I guess it was trying repeatedly in the background to apply the update?).

So the log files not being user-writable were preventing the update from occurring? I'm not sure how the owner of those files got changed...I certainly don't think that's something I did.

**Comment by deepak1556** [comment]
Oh nice!

> After a few seconds VSCode opened with the update applied with no other action from me; I guess it was trying repeatedly in the background to apply the update?

The updating task is handled by https://github.com/Squirrel/Squirrel.Mac framework, and once an update is downloaded it will try to apply it whenever the application quits. So yeah it must have been re-trying it.

> So the log files not being user-writable were preventing the update from occurring? 

This is new, I would expect squirrel to throw some sort of error for this on the handlers we listen to. I will try to setup a minimal example and confirm the issue.

> I'm not sure how the owner of those files got changed

Maybe shipit was owned by root previously ?

**Comment by deepak1556** [comment]
Should have better error signals for this case of failure once https://github.com/electron/electron/pull/26358 lands in upstream.

**Comment by nirmalsinghoo7** [comment]
I am also facing the issue on Mac. I got this alert while while click on Check for Updates - 
![image](https://user-images.githubusercontent.com/12962402/167348930-9c0abff1-7f21-4eca-85ec-ce48a7afa6d3.png)

I have downloaded this install - https://code.visualstudio.com/docs/?dv=darwinarm64 for MacBook Pro M1 Max.

While downloading, it's showing .zip but after downloaded, showing as Visual Studio Code.app

When I click on this file, it's auto started the VS code, not showing any options to install it using drag&drop in Application folder.
